Add flag to log-and-continue on producer errors (#2414)
confluentinc/ksql: docs/developer-guide/processing-log.rst

message.type (INT)
  An int that describes the type of the log message. Currently, the following types are

>   defined: 0 (DESERIALIZATION_ERROR), 1 (RECORD_PROCESSING_ERROR), 2 (PRODUCTION_ERROR).

Reviewer: Can you update the DDL on line 198?

Author: Done -- good catch! Also added the new flag to the config reference docs (and fixed a few errors while I was in there).

message.deserializationError.recordB64 (STRING)
  The Kafka record, encoded in Base64.

message.recordProcessingError (STRUCT)
  The contents of a message with type 1 (RECORD_PROCESSING_ERROR). Logged when KSQL hits
  an error when processing a record, for example, an unexpected null value when evaluating
  an operator in a SELECT clause.

message.recordProcessingError.errorMessage (STRING)
  A string containing a human-readable error message detailing the error encountered.

message.recordProcessingError.record (STRING)
  The KSQL record, serialized as a JSON string.

message.productionError (STRUCT)
  The contents of a message with type 2 (PRODUCTION_ERROR). Logged when a producer fails to

>   publish a Kafka record. These messages will only be logged if ``ksql.fail.on.production.error=false``.

Reviewer 2: ```suggestion
  publish a Kafka record. These messages are logged only if ``ksql.fail.on.production.error=false``.
```

Author: Whoops -- you actually caught a bug. This sentence was true in an earlier iteration of this PR but the most recent version logs error messages to the record processing log regardless of the value of the `ksql.fail.on.production.error` flag, so I've deleted the sentence.
